### PR TITLE
Authorization class

### PR DESCRIPTION
**Pull request by nicolasdanelon**
Session & login management work in progress
getToken, tokenOk,  check_auth (now checkAuth) , isLoggedIn, logout

### CODE REVIEW

**Comment by ArthurHoaro** [comment]
:+1: 

I know it's in progress, but a few thoughts:

  * `setup_login_state()` has its place here.
  * remove that `global` variable.
  * unit tests.

**Comment by nicolasdanelon** [comment]
@ArthurHoaro thanks for the feedback, I need that point of view.

**Comment by nicolasdanelon** [comment]
Oh, my methods are static, is that ok?

**Comment by virtualtam** [comment]
Thanks for the PR!

For all refactoring efforts, I'd appreciate that we use #324 as a hub issue to plan tasks and track the overall progress; this would have the benefits of:
- helping contributors to get started on _what_ to work on and _where_ to look for information;
- helping maintainers to review contributions.

Some feedback on the code:
- _explicit is better than implicit_: explicit class, method and attribute names are preferred, e.g. `Authentication` instead of `Auth`
- session-related methods should be defined in a `Session` class (as roughly planned in #324)
- static methods are OK:
    - they help grouping generic functions (namespacing has not been planned yet, and implies dropping 5.3 support)
    - if needed, the scope can be changed later (e.g. when using proper object instances)

**Comment by nicolasdanelon** [comment]
ok, so let's go to #324 :)

File: tests/AuthenticationTest.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+class AuthenticationTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * Test testLoginState: 
+     * Valid: 
+     *
+     * @return void
+     */
+    public function testLoginStateValid($globals)
```
Comment (ArthurHoaro) [inline tests/AuthenticationTest.php]: UT methods don't take parameters. Call `Authentication::loginState($globals)` inside the method.

**Comment by nicolasdanelon** [comment]
don't have time to work on this. sorry

**Comment by ArthurHoaro** [comment]
No problem, everybody gives the time he has. Let us know if you need help though. :)

**Comment by nicolasdanelon** [comment]
@ArthurHoaro will be awesome some help with test

**Comment by nicolasdanelon** [comment]
@virtualtam only `getToken()` and `tokenOk()` will be in `Session` class, right?

**Comment by nicolasdanelon** [comment]
ok, thanks!

**Comment by nicolasdanelon** [comment]
i'm kinda lost with the new changes.. consider reject this pl or cherry pick what you want :/

**Comment by ArthurHoaro** [comment]
Alright, closing. This still could use a good refactoring, but starting from the current version could be a good idea.

**Pull request closed without merging**